core/src/main/java/io/grpc/internal/GrpcUtil.java

```diff
@@ -431,54 +436,41 @@ public void close(ScheduledExecutorService instance) {
    */
   @VisibleForTesting
   static class TimeoutMarshaller implements Metadata.AsciiMarshaller<Long> {
+    private static final BiMap<TimeUnit, Character> UNITS =
```

Comment:
This should be `ImmutableBiMap` instead of `BiMap` with a comment that it depends on the `ImmutableBiMap` iteration order.